### ISSUE TITLE
unable to use experimental language formatters

### ISSUE THREAD

**Issue by josharian** [tweag/topiary#663]
**Describe the bug**

I'm here kicking the tires, because I need to write a formatter and I love the general approach.

I followed the readme and built everything locally, but can't get any of the experimental language formatters to work. It rejects the language.

**To Reproduce**

Run:

```
$ topiary config | grep -C 3 tree
extensions = ["toml"]

[[language]]
name = "tree_sitter_query"
extensions = ["scm"]
$ topiary fmt -l tree_sitter_query -q /Users/josh/x/topiary/queries/tree-sitter-query.scm go.scm
error: invalid value 'tree_sitter_query' for '--language <LANGUAGE>'
  [possible values: json, nickel, ocaml, ocaml-interface, ocamllex, toml]

For more information, try '--help'.
```

**Expected behavior**

The language is in the config, and I provided a query file, so according to the readme, I think it ought to work?

**Environment**
 - OS name + version: macOS 14.1.2 (23B92)
 - Version of the code: 8cc9aa40d99c89b448190bb1693777128fdcd6c0

**Comment by josharian** [comment]
Thanks. I'm afraid that that workaround doesn't currently work:

```
$ topiary fmt -q /Users/josh/x/topiary/topiary-queries/queries/tree-sitter-query.scm go.scm 
error: The following required argument was not provided: language

Usage: topiary-cli [OPTIONS] <COMMAND>

For more information, try '--help'.
```

I patched in #643, also no luck:

```
$ topiary fmt -l tree_sitter_query -q /Users/josh/x/topiary/topiary-queries/queries/tree-sitter-query.scm queries/go.scm
error: the argument '--language <LANGUAGE>' cannot be used with '[FILES]...'

Usage: topiary format --query <QUERY> <--language <LANGUAGE>|FILES>

For more information, try '--help'.
$ topiary fmt -q /Users/josh/x/topiary/topiary-queries/queries/tree-sitter-query.scm queries/go.scm 
error: The following required argument was not provided: language

Usage: topiary-cli [OPTIONS] <COMMAND>

For more information, try '--help'.
```

I would also note, FWIW, that I found this bit: `<--language <LANGUAGE>|FILES>` pretty hard to parse. I would suggest `<FILES | --language LANGUAGE>` as easier on the eyes/brain.

**Comment by Xophmeister** [comment]
> Thanks. I'm afraid that that workaround doesn't currently work:
> 
> ```
> $ topiary fmt -q /Users/josh/x/topiary/topiary-queries/queries/tree-sitter-query.scm go.scm 
> ```

You don't need to provide the `--query` argument if you edit the default query for the grammar. Topiary will simply pick up those changes transparently; i.e., `topiary fmt go.scm` should do what you expect. Apologies if this wasn't clear.

> I patched in #643, also no luck:

#643 is currently a draft PR; it's not complete yet, but the overall intention of the PR is to decouple the library and CLI, which is where the notion of "supported languages" originates. I therefore suspect there's a good chance that this oversight will eventually be corrected by this PR.

> I would also note, FWIW, that I found this bit: `<--language <LANGUAGE>|FILES>` pretty hard to parse. I would suggest `<FILES | --language LANGUAGE>` as easier on the eyes/brain.

I agree. Unfortunately, this is generated automatically as part of the argument parsing library we use.

**Comment by josharian** [comment]
> `topiary fmt go.scm` should do what you expect.

Hah. Thanks, that did it. Now to go muck around a bit under the hood...

**Comment by Xophmeister** [comment]
Reopening, as this is still a bug that needs fixing.

**Comment by Xophmeister** [comment]
When the feature gating was implemented, there was a feature flag for `experimental` languages. I think you're right that, now dynamic loading has landed, there's no distinction :+1:
